src/common/helper.js

```diff
@@ -389,15 +388,25 @@ async function deleteGroup(tx, group) {
   return groupsToDelete
 }
 
-async function initiateCache() {
-  cache = new nodeCache({
-    stdTTL: 432000,
-    checkperiod: 518400
-  });
+async function acquireRedisClient() {
+  if (redisClient == null) {
+    logger.debug("Creating new redis client")
+    return createRedisClient()
+  } else {
+    const pong = await redisClient.ping()
+    if (!pong) {
+      logger.debug("Redis connection lost, creating new redis client")
+      return createRedisClient()
+    }
+  }
 }
 
-async function getCacheInstance() {
-  return cache
+async function createRedisClient() {
+  const con = redis.createClient({ url: config.REDIS_URL })
+  con.on("error", function (err) {
+    logger.error(err)
+  })
+  return con
```

Comment:
1. Unlike database connections, Redis connection should be established at service start and kept alive instead of connecting and disconnecting each time.
2. Reconnect strategy should be used instead of manual pings.
3. We can have single function here basically creates a client if it is null and connect and return it.